test/test_go.rb

```diff
@@ -1831,6 +1824,20 @@ def test_ctrl_r
     tmux.until { |lines| lines[-1] == '3rd' }
   end
 
+  def test_ctrl_r_abort
+    skip "doesn't restore the original line when search is aborted pre Bash 4" if self.is_a?(TestBash) && /(?<= version )\d+/.match(`#{Shell.bash} --version`).to_s.to_i < 4
+    # Why is this necessary? To wait until the shell is ready?
+    tmux.send_keys 'echo hello', :Enter
+    tmux.until { |lines| lines[-1] == 'hello' }
```

Comment:
Yeah, I guess. You probably should use `tmux.prepare` instead, which also retries for a few time. It's used all over the test cases.

Reply:
Thanks! I'm skeptical about that method in this context: It might add enough delay for the test to succeed, but is keystrokes echoed to the terminal confirmation that the shell is ready? Then again I don't 100% understand why this test doesn't succeed straight up. And if the line subsequently starts with `foo`, that's confirmation that <kbd>C-u</kbd> was handled by the shell, so it's ready. I'll give it a try! :+1:

Reply:
> is keystrokes echoed to the terminal confirmation that the shell is ready?

Maybe it's just better than nothing? But I agree that it's not sufficient. `git blame` on the method shows that we used to send `Right` key as well (https://github.com/junegunn/fzf/commit/400e443a0a397e559250fa090b5c437e44d01bf8), but for some reason I don't quite remember I dropped it. It makes more sense to have it because if the shell is not ready, the command line will print something like `foo^[[C`, so the subsequent `end_with?` test will fail.

So I just pushed https://github.com/junegunn/fzf/commit/2ff7db1b36e4f976e4e24e56e996b9128f131141, which hopefully works better.

Reply:
> that's confirmation that C-u was handled by the shell, so it's ready.

Yep, that was the intention. But it looks like CTRL-U can be handled even when the shell is not available, i.e. `sleep 60` and `foo<CTRL-U>` just works. I don't know if something has changed over the years. But anyway, cursor keys still leaves `^[[D` or `^[[C` on the screen, so I believe 2ff7db1 should definitely help.